src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableOrchestrationContext.cs

```diff
@@ -223,12 +223,13 @@ public interface IDurableOrchestrationContext
         /// </remarks>
         /// <param name="name">The name of the event to wait for.</param>
         /// <param name="timeout">The duration after which to throw a TimeoutException.</param>
+        /// <param name="cancelToken">The <c>CancellationToken</c> to use for cancelling <paramref name="timeout"/>'s internal timer.</param>
         /// <typeparam name="T">Any serializeable type that represents the JSON event payload.</typeparam>
         /// <returns>A durable task that completes when the external event is received.</returns>
         /// <exception cref="TimeoutException">
         /// The external event was not received before the timeout expired.
         /// </exception>
-        Task<T> WaitForExternalEvent<T>(string name, TimeSpan timeout);
+        Task<T> WaitForExternalEvent<T>(string name, TimeSpan timeout, CancellationToken cancelToken);
```

Comment:
I think we need to include a default parameter here to prevent this from being a breaking change in production code.

Reply:
Are the extension methods I included insufficient?

Reply:
Ah you are totally right. The main concern now is that this change will break tests that were mocking this method, as Moq is unable to mock an extension method. I think this is fine as long as we call it out in the release notes.

@cgillum, thoughts?

Reply:
I'm okay with this change since it only impacts tests, but we'll definitely need to call it out as a breaking change in the release notes.

Reply:
Yep, don't think there's any way around that one.  I also ran into this very issue with one of the built-in unit tests.  In my experience, issues like these are easy to recover from. :smile:

Reply:
Is there anything I can add to help make sure this gets into release notes, or is that something you manage outside of the repo?  I don't see an in-repo changelog, so I assume the latter, but am happy to include something extra here if it'd help.

Reply:
We generally just look at the change log from the previous release and construct the release notes from that, though I am not opposed to maintaining something as we merge PRs. It would simplify the work we have to do at release time.

Reply:
In that case, I don't want to start a new process for you all, so I'll let the normal process continue.  Happy to adapt in case you decide to change this for the future!